lib/src/resolve-symlinks.js

```diff
@@ -13,14 +13,22 @@ function resolveSymlinks(opt) {
 
   // A stat property is exposed on file objects as a (wanted) side effect
   function resolveFile(globFile, enc, callback) {
+    var stat = opt.statCache[globFile.path];
+    if (stat && (resolveSymlinks !== stat.isSymbolicLink())) {
```

Comment:
I think I see what you mean. I've been working with a branch of node-glob that would let me further reduce some fs calls, and with that branch it would be possible for the first stat we get from the cache to be symbolic. That's why this test currently catches both ways.

I'm not holding my breath on node-glob/minimatch PRs though, unfortunately the maintainer seems a little bit too busy to address issues. Some of which strike me as pretty serious (i.e. complete mismatch of path separator handling between glob and minimatch on windows) and I wonder if it might make sense to proceed with a fork of those libraries...